src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Config/SetAzureVMOperatingSystemCommand.cs

```diff
@@ -238,7 +303,18 @@ public override void ExecuteCmdlet()
                 }
 
                 // OS Profile
-                this.VM.OSProfile.WindowsConfiguration.ProvisionVMAgent = this.ProvisionVMAgent.IsPresent;
+                if (this.ProvisionVMAgent.IsPresent)
+                {
+                    this.VM.OSProfile.WindowsConfiguration.ProvisionVMAgent = true;
+                }
+                else if (this.DisableVMAgent.IsPresent)
+                {
+                    this.VM.OSProfile.WindowsConfiguration.ProvisionVMAgent = false;
+                }
+                else
+                {
+                    this.VM.OSProfile.WindowsConfiguration.ProvisionVMAgent = this.VM.OSProfile.WindowsConfiguration.ProvisionVMAgent;
```

Comment:
@hyonholee is this `else` block necessary? It looks like we just need the `if` and `else-if` blocks